pkg/controller/provider/controller.go

```diff
@@ -437,3 +420,28 @@ func (r *Catalog) get(request reconcile.Request) (p *api.Provider, found bool) {
 	p, found = r.content[request]
 	return
 }
+
+func (r *Reconciler) CleanupProviderOnDelete(provider *api.Provider) (err error) {
```

Comment:
we should change the visibility of this function and the name should reflect that it only handles OVA providers
```suggestion
func (r *Reconciler) removeVolumeOfOVAServer(provider *api.Provider) (err error) {
```

Reply:
@ahadas open a separate PR for the refactoring https://github.com/kubev2v/forklift/pull/834.